Upmerge 14.12.2020 Release Notes Update (#3682)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-latest.rst

* Optimized ROM and RAM used by Thread samples.
* Disabled Hardware Flow Control on the serial port in :ref:`coap_client_sample` and :ref:`coap_server_sample` samples.

> * Development support for nRF21540 FEM (front-end module) in the Thread samples.

Reviewer: Please move up to line 73 and indent - this is not an update, but something that has been added.

  * Time client model callbacks for all message types.
  * Support for the nRF52833 DK in the :ref:`bluetooth_mesh_light` and :ref:`bluetooth_mesh_light_switch` samples.

>   * Added support for Light xyL models.
>   * Disable light control server when manual control has taken effect. This follow Mesh Model Specification section 6.2.3.

Reviewer: ```suggestion
  * Support for :ref:`bt_mesh_light_xyl_readme`.
  
* Updated the :ref:`bt_mesh_lightness_srv_readme` to disable the light control server when manual control has taken effect. 
  This follows the Mesh Model Specification section 6.2.3.
```

See the changelog for each library in the :doc:`nrfxlib documentation <nrfxlib:README>` for the most current information.

Multiprotocol Service Layer (MPSL)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> * Modified the nRF5340 to use SWI by default, instead of EGU.

Reviewer: What have we modified here? Surely not the nRF5340 SoC, which is what we're saying at the moment. ;)

> * Modified the Bluetooth controller for nRF5340 to use SWI by default, instead of EGU.

Reviewer: This is about https://github.com/zephyrproject-rtos/zephyr/pull/27801, correct?
Why is this related to MPSL? To me it looks like an update to the Zephyr Bluetooth Controller.
So it should be moved down to the Zephyr > Drivers > Bluetooth Controller section.
Text suggestion:
Updated the Bluetooth Controller to use SWI instead of EGU for nRF5340.

Author: Leaving this to Bluetooth people.

* Added support for invoking shell commands (shell management) from the mcumgr command line.

> Multiprotocol Service Layer (MPSL)

Reviewer: MPSL is part of sdk-nrfxlib, so this section should be moved up (line 172/177) and made a subsection (~~~)

==================================

> * The nRF53 has been modified to use SWI by default, instead of EGU.

Reviewer: ```suggestion
* Modified the nRF5340 to use SWI by default, instead of EGU.
```
What exactly was updated here? MPSL support for nRF5340?
I doubt we updated the chip. ;-)

    * Added multiple ``nrfx_power``-related fixes to reduce power consumption.

  * PWM:

>     * Changed GPIO configuration to Nordic HAL which allows support for GPIO  pins above 31.
>     * Added check if PWM period does not exceed 16-bit value to prevent broken behaviour.
>     * Changed PWM DT configuration to use timer phandle instead of previously used timer-instance.

Reviewer: ```suggestion
    * Changed the GPIO configuration to use Nordic HAL, which allows support for GPIO pins above 31.
    * Added a check to ensure that the PWM period does not exceed a 16-bit value to prevent erroneous behavior.
    * Changed the PWM DT configuration to use a timer phandle instead of the previously used timer instance.
```

    * Deprecated legacy TCP stack (TCP1).
    * Added multiple minor TCP2 bugfixes and improvements.
    * Added network management events for DHCPv4.

>     * Fixed getaddrinfo to actually respect socket type hints.

Reviewer: Is this covered by https://github.com/nrfconnect/sdk-nrf/pull/3647/files line 484?

      * :option:`CONFIG_OPENTHREAD_MAC_SOFTWARE_ACK_TIMEOUT_ENABLE`
      * :option:`CONFIG_OPENTHREAD_MAC_SOFTWARE_RETRANSMIT_ENABLE`
      * :option:`CONFIG_OPENTHREAD_PLATFORM_USEC_TIMER_ENABLE`

>       * :option:`CONFIG_OPENTHREAD_PLATFORM_INFO`

Reviewer: Almost duplicating https://github.com/nrfconnect/sdk-nrf/pull/3647/files

Author: Unfortunately I do not understand.

Reviewer: Some of these options are also added in PR https://github.com/nrfconnect/sdk-nrf/pull/3647
I don't mind where they are added, but adding them only once would probably prevent merge conflicts. ;)

Author: > 
> 
> Some of these options are also added in PR #3647
> I don't mind where they are added, but adding them only once would probably prevent merge conflicts. ;)

Ah... right. My bad.
